renku/ui/cli/dataset.py

```diff
@@ -627,40 +630,84 @@ def create(name, title, description, creators, metadata, keyword):
 
 @dataset.command()
 @click.argument("name", shell_complete=_complete_datasets)
-@click.option("-t", "--title", default=None, type=click.STRING, help="Title of the dataset.")
-@click.option("-d", "--description", default=None, type=click.STRING, help="Dataset's description.")
+@click.option("-t", "--title", default=NO_VALUE, type=click.UNPROCESSED, help="Title of the dataset.")
+@click.option("-d", "--description", default=NO_VALUE, type=click.UNPROCESSED, help="Dataset's description.")
 @click.option(
     "-c",
     "--creator",
     "creators",
-    default=None,
+    default=[NO_VALUE],
     multiple=True,
+    type=click.UNPROCESSED,
     help="Creator's name, email, and affiliation. " "Accepted format is 'Forename Surname <email> [affiliation]'.",
 )
 @click.option(
     "-m",
     "--metadata",
-    default=None,
-    type=click.Path(exists=True, dir_okay=False),
+    default=NO_VALUE,
+    type=click.UNPROCESSED,
     help="Custom metadata to be associated with the dataset.",
 )
-@click.option("-k", "--keyword", default=None, multiple=True, type=click.STRING, help="List of keywords or tags.")
-def edit(name, title, description, creators, metadata, keyword):
+@click.option(
+    "-k",
+    "--keyword",
+    "keywords",
+    default=[NO_VALUE],
+    multiple=True,
+    type=click.UNPROCESSED,
+    help="List of keywords or tags.",
+)
+@click.option(
+    "-u",
+    "--unset",
+    default=[],
+    multiple=True,
+    type=click.Choice(["keywords", "k", "images", "i", "metadata", "m"]),
+    help="Remove keywords from dataset.",
+)
+def edit(name, title, description, creators, metadata, keywords, unset):
     """Edit dataset metadata."""
     from renku.command.dataset import edit_dataset_command
     from renku.core.util.metadata import construct_creators
     from renku.ui.cli.utils.callback import ClickCallback
 
-    creators = creators or ()
-    keywords = keyword or ()
+    images = NO_VALUE
 
-    custom_metadata = None
+    if list(creators) == [NO_VALUE]:
+        creators = NO_VALUE
+
+    if list(keywords) == [NO_VALUE]:
+        keywords = NO_VALUE
+
+    if "c" in unset or "creators" in unset:
```

Comment:
`unset` doesn't accept creators.

Reply:
Right, I had that initially but then decided against allowing no creators